openwisp_firmware_upgrader/tests/test_api/test_views.py

```diff
@@ -0,0 +1,534 @@
+from django.test import Client, TestCase
+from django.urls import reverse
+from openwisp_firmware_upgrader.api.serializers import (
+    BatchUpgradeOperationSerializer,
+    BuildSerializer,
+    CategorySerializer,
+    FirmwareImageSerializer,
+)
+from openwisp_firmware_upgrader.tests.test_api.base import TestAPIUpgraderMixin
+from swapper import load_model
+
+BatchUpgradeOperation = load_model('firmware_upgrader', 'BatchUpgradeOperation')
+Build = load_model('firmware_upgrader', 'Build')
+Category = load_model('firmware_upgrader', 'Category')
+DeviceFirmware = load_model('firmware_upgrader', 'DeviceFirmware')
+FirmwareImage = load_model('firmware_upgrader', 'FirmwareImage')
+UpgradeOperation = load_model('firmware_upgrader', 'UpgradeOperation')
+
+
+class TestBuildViews(TestAPIUpgraderMixin, TestCase):
+    batch_upgrade_operation_model = BatchUpgradeOperation
+    build_model = Build
+    category_model = Category
+    device_firmware_model = DeviceFirmware
+    firmware_image_model = FirmwareImage
+    upgrade_operation_model = UpgradeOperation
+
+    def _serialize_build(self, build):
+        serializer = BuildSerializer()
+        return dict(serializer.to_representation(build))
+
+    def test_build_unauthorized(self):
+        build = self._create_build()
+        client = Client()
```

Comment:
these tests should go in `tests/tst_api/base/test_views.py` and be imported here as well as in the sample_app tests.
Applies for the other test classes as well.